tests/unit/package_managers/yarn_classic/test_utils.py

```diff
@@ -0,0 +1,69 @@
+from unittest import mock
+
+from cachi2.core.package_managers.yarn_classic.project import PackageJson
+from cachi2.core.package_managers.yarn_classic.utils import find_runtime_deps
+from cachi2.core.rooted_path import RootedPath
+
+PACKAGE_JSON_CONTENT = """
+{
+  "name": "project",
+  "dependencies": {
+    "main-dep1": "^1.0.0"
+  },
+  "optionalDependencies": {
+    "optional-dep1": "^2.0.0"
+  },
+  "peerDependencies": {
+    "peer-dep1": "^3.0.0"
+  },
+  "devDependencies": {
+    "dev-dep1": "^4.0.0"
+  }
+}
+"""
+
+
+@mock.patch("cachi2.core.package_managers.yarn_classic.project.YarnLock")
+@mock.patch("cachi2.core.package_managers.yarn_classic.utils._expand_yarn_lock_keys")
+def test_find_runtime_deps(
+    mock_expand_yarn_lock_keys: mock.Mock,
+    mock_yarn_lock: mock.Mock,
+    rooted_tmp_path: RootedPath,
+) -> None:
+    rooted_tmp_path.join_within_root("package.json").path.write_text(PACKAGE_JSON_CONTENT)
+    package_json = PackageJson.from_file(rooted_tmp_path.join_within_root("package.json"))
+
+    mock_expand_yarn_lock_keys.return_value = {
+        # dependencies
+        "main-dep1@^1.0.0": {
+            "version": "1.0.0",
+            "dependencies": {"sub-dep1": "^1.1.0"},
+        },
+        # optional dependencies
+        "optional-dep1@^2.0.0": {
+            "version": "2.0.0",
+            "dependencies": {"multi-dep1": "^4.0.0", "multi-dep2": "^4.0.0"},
+        },
+        "multi-dep1@^4.0.0, multi-dep2@^4.0.0": {"version": "5.0.0", "dependencies": {}},
+        # peer dependencies
+        "peer-dep1@^3.0.0": {"version": "3.0.0", "dependencies": {}},
+        # transitive dependencies
+        "sub-dep1@^1.1.0": {"version": "1.1.0", "dependencies": {"sub-dep11": "^1.1.1"}},
+        "sub-dep11@^1.1.1": {"version": "1.1.1", "dependencies": {}},
+        # dev dependencies
+        "dev-dep1@^4.0.0": {"version": "4.0.0", "dependencies": {}},
+    }
+
+    result = find_runtime_deps(package_json, mock_yarn_lock, [])
```

Comment:
Should there be a test for `Note: If a dependency is marked as runtime dependency somewhere and as a development dependency somewhere else, it is classified as runtime`? E.g. by having a simple workspace where `dev-dep1` is a runtime dependency?